base/leaky_bucket.go

```diff
@@ -283,7 +283,12 @@ func (b *LeakyBucket) UUID() (string, error) {
 		return b.bucket.UUID()
 }
 
-
+func (b *LeakyBucket) CloseAndDelete() error {
+	if bucket, ok := b.bucket.(sgbucket.DeleteableBucket); ok {
+		return bucket.CloseAndDelete()
+	}
+	return nil
+}
```

Comment:
Originally a failing unit test, failed locally and on drone, can't repro locally now, have pushed with method commented out to see if drone passes as well. Depending on drone result I can remove.